landscaper/pkg/gardenlet/chart/charttest/charttest.go

```diff
@@ -717,7 +717,7 @@ func ComputeExpectedGardenletConfiguration(
 		SNI: &gardenletconfigv1alpha1.SNI{Ingress: &gardenletconfigv1alpha1.SNIIngress{
 			ServiceName: pointer.StringPtr(gardenletconfigv1alpha1.DefaultSNIIngresServiceName),
 			Namespace:   pointer.StringPtr(gardenletconfigv1alpha1.DefaultSNIIngresNamespace),
-			Labels:      map[string]string{"istio": "ingressgateway"},
+			Labels:      map[string]string{"app": "istio-ingressgateway", "istio": "ingressgateway"},
```

Comment:
Why is the label `"istio": "ingressgateway"` used here while it seemed to be removed everywhere else?

Reply:
The `"istio": "ingressgateway"` is label is kept for compatibility reasons. I don't want to break existing ingress gateway e.g. changing the selector of the envoy/ingress gateway deployment is not possible as it is immutable. 

The default ingress gateway (`istio-ingress` namespace) received those label in the past from the `istio/ingress-gateway` chart where it is a default value. I removed the label from the charts default values to allow reuse of the chart without label clashes for other exposureclass handler ingress gateways.

Reply:
As said I would rather vote for removing the ability to set custom labels for the ingress gateways at all.
Consequently we should remove the `.sni.ingress.labels` and `.exposureClassHandler[].sni.ingress.labels` in the GardenletConfiguration. 
For me is not clear why we need that at all. In the end you can configure there label clashes and wrong ingress gateways could used to serve kube-apiservers which should be backed by another ingress gateway.
cc @mvladev